arangod/Aql/LimitExecutor.cpp

```diff
@@ -75,7 +75,7 @@ auto LimitExecutor::calculateUpstreamCall(AqlCall const& clientCall) const -> Aq
 
   // Generally, we create a hard limit. However, if we get a soft limit from
   // downstream that is lower than our hard limit, we use that instead.
```

Comment:
Maybe we should adjust that comment now, too.

Reply:
No the comment is still correct.
We check for not hardLimit.
The reason here is that getLimit() considers the batchSize, so if you have a high softLimit, but suing a small batchSize this will actually convert to hardLimit of batch size.